enhancements/workload-partitioning/management-workload-partitioning-api.md

```diff
@@ -0,0 +1,249 @@
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The overall feature summary is present in the [Management workload partitioning](workload-partitioning/management-workload-partitioning.md) feature. This document focuses on how the feature enablement API and its interactions will look like.
+
+## Motivation
+
+This feature is about designing a read-only API that will describe the enabled workload partitions (types, classes, etc.). This information is needed for kubelet to start exposing the right resources as well for the admission webhook to know when the pod manipulation is needed.
+
+### Goals
+
+The short term goal is to get enough functionality to create the `management` partition where all OpenShift infrastructure could be placed. This partition is cluster wide and only affects pods with certain pre-agreed-upon labels and placed in specific pre-annotated namespaces.
+
+### Non-Goals
+
+- Long term goal of allowing cluster admins to create their own partitions is not part of this proposal for now
+- Day 2 creation of partitions is not part of the proposal
+- Partitions with scope less than the entire cluster (such as being limited to Namespaces or MCPs) are not part of the proposal
+
+## Proposal
+
+The proposal is to define a new cluster-wide Custom Resource Definition that would describe the allowed partition names in the status section. That way it hints at being a read only object where no user/admin input or modifications are expected.
+
+```yaml
+apiVersion: workload.openshift.io/v1
+kind: WorkloadPartitions
+metadata:
+  # arbitrary name, all objects of this Kind should be processed and merged
+  name: management-partition
+status:
+  # List of strings, defines partition names that will be recognized by the
+  # workload partitioning webhook. This list will also inform PAO about partitions
+  # that should be configured on the kubelet and CRI-O level.
+  clusterPartitionNames:
+    - management
+```
+
+It is expected this API will be created at the installation process. Either manually or using the Performance Addon Operator render mode.
+
+To allow for future extensibility and possible multiple sources of workload partition names (coming from customers, the installer, or other operators, etc.), we propose that there might be multiple `WorkloadPartitions` objects injected into the cluster. The expected behavior is that all components would just merge all the defined names together.
+
+There is no controller or reconcile loop as part of this proposal. Only the cluster administrator will have the ability to create or manipulate the WorkloadPartitions objects. Anyone will be allowed to read them.
```

Comment:
We still will run reconciliation under the PAO once the `WorkloadPartitions` resource updated.